### PR TITLE
[GCU E2E Testing] Update test_update_speed and test_replace_fec to consider StateDB validator

### PR DESCRIPTION
**Pull request by isabelmsft**
ADO 24806590

### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
StateDB Validator was added via this PR: https://github.com/sonic-net/sonic-utilities/pull/2936/files
GCU E2E Testing for port speed and fec needs to be updated to account for new StateDB Validator
#### How did you do it?
Update GCU E2E test cases to validate against StateDB before determining whether to expect success or failure
#### How did you verify/test it?
Run E2E test on physical DUT

### CODE REVIEW

File: tests/generic_config_updater/test_eth_interface.py [hunk 1]
```diff
@@ -294,7 +321,7 @@ def test_update_speed(duthost, ensure_dut_readiness):
 
         try:
             output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
-            if is_valid:
+            if is_valid and is_valid_speed_state_db(duthost, speed):
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_eth_interface.py:324]: >is_valid_speed_state_db [](http://example.com/codeflow?start=28&length=23)

Did you test one speed which is allowed by sonic yang model, but not valid in StateDB?

**Comment by qiluo-msft** [comment]
> @pytest.mark.parametrize("fec", ["rs", "fc"])

Did you test one fec which is allowed by sonic yang model, but not valid in StateDB?

---
Refers to: tests/generic_config_updater/test_eth_interface.py:228 in b1d09ed. [](commit_id = b1d09ed9a101de6fb2314ab8a8e2c1796bd159c1, deletion_comment = False)

File: tests/generic_config_updater/test_eth_interface.py [hunk 2]
```diff
@@ -37,6 +37,30 @@ def ensure_dut_readiness(duthost):
         delete_checkpoint(duthost)
 
 
+def is_valid_fec_state_db(duthost, value):
+    read_supported_fecs_cli = 'redis-cli -n 6 hget "PORT_TABLE|{}" supported_fecs'.format("Ethernet0")
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_eth_interface.py:41]: >redis-cli -n 6 [](http://example.com/codeflow?start=31&length=14)

sonic-db-cli STATE_DB

**Comment by isabelmsft** [comment]
> > @pytest.mark.parametrize("fec", ["rs", "fc"])
> 
> Did you test one fec which is allowed by sonic yang model, but not valid in StateDB?
> 
> Refers to: tests/generic_config_updater/test_eth_interface.py:228 in [b1d09ed](https://github.com/sonic-net/sonic-mgmt/commit/b1d09ed9a101de6fb2314ab8a8e2c1796bd159c1). [](commit_id = [b1d09ed](https://github.com/sonic-net/sonic-mgmt/commit/b1d09ed9a101de6fb2314ab8a8e2c1796bd159c1), deletion_comment = False)

Yes, I updated the code to test all fec allowed by yang model: "rs", "fc", "none." So by default, where one mode is allowed by yang model but not StateDB, this scenario is tested

File: tests/generic_config_updater/test_eth_interface.py (hunk 2 above)
Comment (isabelmsft) [reply tests/generic_config_updater/test_eth_interface.py]: Changed to `sonic-db-cli STATE_DB`

File: tests/generic_config_updater/test_eth_interface.py (hunk 1 above)
Comment (isabelmsft) [reply tests/generic_config_updater/test_eth_interface.py:328]: Yes thank you, I updated the code to test this scenario

**Review by qiluo-msft — DISMISSED** [review]

**Review by qiluo-msft — DISMISSED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Review by wen587 — APPROVED** [review]

**Comment by StormLiangMS** [comment]
hi @isabelmsft this PR will make 202305 test failure, I will revert, pls test with 202305 image and submit separate PR to 202305

**Pull request merged**